aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -186,6 +186,72 @@ Tensor ger(const Tensor& self, const Tensor& vec2) {
   return self.outer(vec2);
 }
 
+Tensor& inner_out(Tensor& out, const Tensor& self, const Tensor& other) {
+  checkDeviceType("inner()", {out, self, other}, self.device().type());
```

Comment:
Wait a minute... isn't `inner` = `np.tensordot(a, b, axes=(-1,-1))` => `torch.tensordot(a, b, dims=((-1,), (-1,)))`?

Reply:
torch.inner(a, b) == torch.tensordot(a, b, dims=([-1], [-1])) for non-scalar a and b only. For scalars, it behaves differently.

Reply:
OK, so we're halfway there. What happens for scalars? Can we write the op as a composite of tensordot plus some scalar handling?

Reply:
Actually we can, torch.tensordot(a, b, dims=0) is equivalent to torch.inner if either a or b are scalars.

Reply:
On second thought, why replace a multiplication by a call to tensordot? Same number of lines but worse performance.